src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -2,34 +2,123 @@
  */
 public class Elevator extends SubsystemBase {
 
     private final CANSparkMax elevatorMotor = new CANSparkMax(Const.CAN.ELEVATOR_MOTOR, MotorType.kBrushless);
     private final WPI_VictorSPX wheelMotor = new WPI_VictorSPX(Const.CAN.ELEVATOR_WHEEL);
     private final ColorSensorV3 colorSensor = new ColorSensorV3(I2C.Port.kOnboard);
+    private final CANPIDController elevatorPID = new CANPIDController(elevatorMotor);
+    private String targetColor = "";
+    private int revCount = 0;
+    private boolean done = false;
+
 
     public Elevator() {
 
     }
+
+    /**
+     * Position loop control for the elevator.
+     * 
+     * @param position Encoder reference for elevator setpoint.
+     */
+    public void setElevatorPosition(double position) {
+        elevatorPID.setP(Const.PID.ELEVATOR_P);
+        elevatorPID.setI(Const.PID.ELEVATOR_I);
+        elevatorPID.setReference(position, ControlType.kPosition);
+    }
+
+    /**
+     * Velocity loop control for the elevator.
+     * 
+     * @param velocity Speed reference for elevator.
+     */
+    public void setElevatorVelocity(double velocity) {
+        elevatorPID.setP(Const.PID.ELEVATOR_P);
+        elevatorPID.setI(Const.PID.ELEVATOR_I);
+        elevatorPID.setReference(velocity, ControlType.kVelocity);
+    }
+
+    /**
+     * Sets the color wheels speed.
+     * 
+     * @param speed The speed, between -1 and 1, of the motor.
+     */
+    public void setWheelSpeed(double speed) {
+        wheelMotor.set(speed);
+    }
+
+    /**
+     * Spins color wheel to meet color and revolution needs.
+     * 
+     * @param targetRevs Number of revolutions desired.
+     * @param colorMode  Set true if we need to match the color on the wheel.
+     */
+    public void spinColorWheel(int targetRevs, boolean colorMode) {
+        /** Counts revs with every color change. */
+        String color = colorSensor.getColor().toString();
+        if (colorSensor.getColor().toString() != color) {
+            revCount++;
+            color = colorSensor.getColor().toString();
+        }
+        /** Only counting revolutions, no color. */
+        if (!colorMode) {
+            while (revCount < targetRevs * 8) {
+                wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+            }
+        /** Counting revolutions with the color. */
+        } else if (colorMode) {
+            if (color != colorToTargetColor() && !done) {
+                wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+            } else {
+                done = true;
+                while (revCount < targetRevs * 8) {
+                    wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+                }
+            }
+        }
+    }
+
+    /**
+     * Serves only the purpose of finding the offset targeted color for the function
+     * spinColorWheel().
+     */
+    public String colorToTargetColor() {
+        String color = DriverStation.getInstance().getGameSpecificMessage();
```

Comment:
This is not the subsystem's job to get the color from the driver station. We'll do this in a command, and just pass it a `WheelColor` object.